sw/airborne/subsystems/abi_sender_ids.h

```diff
@@ -429,11 +429,21 @@
 #define RELATIVE_LOCALIZATION_ID 1
 #endif
 
-
 #ifndef DETECT_GATE_ABI_ID
 #define DETECT_GATE_ABI_ID 33
 #endif
 
+/*
+ * VISUAL_DETECTION communication (message 27)
+*/
+#ifndef COLOR_OBJECT_DETECTION1_ID
+#define COLOR_OBJECT_DETECTION1_ID 1
```

Comment:
Wouldn't it be more extensible to add an ID field (camera, object or whatever is needed) to the ABI message rather than multiple sender ID ?

Reply:
No. The number of filters is also limited by the inability of the airframe and module settings to handle an arbitrary number of defines/variables. (For instance, right now the filter number is hardcoded into the define name, as in COLOR_OBJECT_DETECTOR_LUM_MIN**1**). This is already causing some code duplication inside the detect_color_object module.

I did some experiments where a single define could set up an arbitrary number of filters. This is technically possible but I ran into some problems (#2454). It would also not provide a solution for the settings in the GCS.

I can move the ID inside the ABI message (in fact, this is already done by the detector) and test for this value inside a single callback, but this would not make the system any more extensible.

Reply:
does my comments for #2454 helps you in any way ?

Reply:
It made it clear why my attempts to fix this did not work, but even if #2454 could be fixed it would still not be possible to add a variable number of settings to the module.
I would rather leave the code as-is for now as this has already been testflown and handling the IDs differently would not really improve much.